kedro/contrib/io/azure/csv_blob.py

```diff
@@ -107,30 +112,80 @@ def __init__(
                 Here you can find all available arguments:
                 https://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.to_csv.html
                 All defaults are preserved, but "index", which is set to False.
+            version: If specified, should be an instance of
+                ``kedro.io.core.Version``. If its ``load`` attribute is
+                None, the latest version will be loaded. If its ``save``
+                attribute is None, save version will be autogenerated.
 
         """
-        self._filepath = filepath
-        self._container_name = container_name
-        self._credentials = credentials if credentials else {}
-        self._blob_to_text_args = blob_to_text_args if blob_to_text_args else {}
-        self._blob_from_text_args = blob_from_text_args if blob_from_text_args else {}
-        super().__init__(load_args, save_args)
+        _container_name = container_name
+        _credentials = copy.deepcopy(credentials)
+        _blob_service = BlockBlobService(**_credentials)
+        glob_function = partial(
+            _glob,
+            blob_service=_blob_service,
+            filepath=filepath,
+            container_name=_container_name,
+        )
+        exists_function = partial(
+            _exists_blob, blob_service=_blob_service, container_name=_container_name
+        )
+
+        super().__init__(
+            load_args=load_args,
+            save_args=save_args,
+            filepath=PurePosixPath(filepath),
+            version=version,
+            exists_function=exists_function,
+            glob_function=glob_function,
+        )
+
+        self._blob_to_text_args = copy.deepcopy(blob_to_text_args) or {}
+        self._blob_from_text_args = copy.deepcopy(blob_from_text_args) or {}
+
+        self._container_name = _container_name
+        self._credentials = credentials
```

Comment:
We should use the deepcopied one.
```suggestion
        self._credentials = _credentials
```